src/test/uint256_tests.cpp

```diff
@@ -330,6 +331,57 @@ BOOST_AUTO_TEST_CASE(parse)
     }
 }
 
+/**
+ * Implemented as a templated function so it can be reused by other classes that have a FromHex()
+ * method that wraps uint256::FromHex(), such as transaction_identifier::FromHex().
+ */
+template <typename T>
+void TestFromHex64Chars()
+{
+    std::string valid_64char_input{"0123456789abcdef0123456789ABCDEF0123456789abcdef0123456789ABCDEF"};
+    {
+        // check that lower and upper case hex characters are accepted
+        auto valid_result{T::FromHex(valid_64char_input)};
+        BOOST_REQUIRE(valid_result);
+        BOOST_CHECK_EQUAL(valid_result->ToString(), ToLower(valid_64char_input));
+    }
+    {
+        // check that only strings of size 64 are accepted
+        BOOST_CHECK(!T::FromHex(""));
+        BOOST_CHECK(!T::FromHex("0"));
+        BOOST_CHECK(!T::FromHex("0123456789abcdef0123456789ABCDEF"));                                  // 32 chars
+        BOOST_CHECK(!T::FromHex("0123456789abcdef0123456789ABCDEF0123456789abcdef0123456789ABCDE"));   // 63 chars
+        BOOST_CHECK(!T::FromHex("0123456789abcdef0123456789ABCDEF0123456789abcdef0123456789ABCDEF0")); // 65 chars
+    }
+    {
+        // check that non-hex characters are not accepted
+        std::string invalid_chars{R"( !"#$%&'()*+,-./:;<=>?@GHIJKLMNOPQRSTUVWXYZ[\]^_`ghijklmnopqrstuvwxyz{|}~)"};
+        std::string valid_base{"0123456789abcdef0123456789ABCDEF0123456789abcdef0123456789ABCDE"}; // 63 chars
+        BOOST_CHECK_EQUAL(valid_base.length(), 63);
+        for (auto c : invalid_chars) {
+            BOOST_CHECK(!T::FromHex(valid_base + c));
+        }
+        // 0x prefixes are invalid
+        std::string invalid_prefix{"0x" + valid_64char_input};
+        BOOST_CHECK(!T::FromHex(std::string_view(invalid_prefix.data(), 64)));
+        BOOST_CHECK(!T::FromHex(invalid_prefix));
+    }
+    {
+        // check that string_view length is respected
+        std::string chars_68{valid_64char_input + "0123"};
+        BOOST_CHECK(T::FromHex(std::string_view(chars_68.data(), 64)));
```

Comment:
I initially changed it to what it is now because I wanted to keep the test focused on ensuring string_view length to be respected, but I suppose it is helpful to ensure the correct substring characters are respected too, so I've taken your suggestion, thanks.